src/node/stage/approved.rs

```diff
@@ -245,6 +272,85 @@ impl Approved {
                 self.send_parsec_poke(core);
                 self.timer_token = core.timer.schedule(KNOWLEDGE_TIMEOUT);
             }
+            let ori_ticks = self.dkg_ticks;
+            if let Some(ticks) = ori_ticks {
+                if ticks > 5 {
+                    self.dkg_ticks = Some(0);
+                    self.progress_dkg(core);
+                } else {
+                    self.dkg_ticks = Some(ticks + 1);
+                }
+            }
```

Comment:
It would be cleaner to have a separate timer token for this stuff. Something like:
```rust
if token == self.gossip_timer_token { // rename `timer_token` to `gossip_timer_token` for clarity
    // ...
} else if token == self.dkg_timer_token {
    self.dkg_timer_token = core.timer.schedule(DKG_PROGRESS_INTERVAL);
    self.progress_dkg(core);
}
```